Tests/TuistDependenciesTests/SwiftPackageManager/Utils/PackageInfoMapperTests.swift

```diff
@@ -2792,16 +2871,26 @@ extension ProjectDescription.Target {
         customSettings: ProjectDescription.SettingsDictionary = [:],
         moduleMap: String? = nil
     ) -> Self {
-        .init(
+        var sources: SourceFilesList?
+
+        switch customSources {
+        case let .some(list) where !list.globs.isEmpty:
+            sources = list
+        case .none:
+            sources = .init(globs: [basePath.appending(RelativePath("Package/Sources/\(name)/**")).pathString])
+        default:
+            break
+        }
```

Comment:
Why do you need this change? can't you just pass nil instead of an empty one from the test?

Reply:
the previous implementation would substitute nil values with a default value, so the resulting target would never have a `nil` source files path

Reply:
I see, but I find it confusing now (i.e. to set a nil source you need to pass an empty source).

What about:
- renaming `customSource` parameter to `source`, which defaults to .init(globs: [basePath.appending(RelativePath("Package/Sources/\(name)/**")).pathString])
- you will need to change the type to be a function that takes a String in input, because the default depends on the target name

Reply:
can you explain the idea with the function a bit more? 

I'm leaning towards an enum-based approach, e.g.

```swift
enum SourceFilesListType {
   case `default`
   case custom(SourceFilesList)
}


fileprivate static func test(..., customSources: SourceFilesListType? = nil) {
  var sources: SourceFilesList?

  switch customSources {
    case let .custom(list):
      sources = list
    case .`default`:
      sources = .init(globs: [basePath.appending(RelativePath("Package/Sources/\(name)/**")).pathString])
    default:
      break
  }
  ...
}
```

Reply:
yep, an enum is fine as well 👌 

Small note: `customSources: SourceFilesListType` shouldn't be nillable, while `custom(SourceFilesList?)` should

Reply:
done